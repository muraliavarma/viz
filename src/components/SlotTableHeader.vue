<script>
export default {
  name: 'SlotTableHeader',
  functional: true,
  render(createElement, ctx) {
    return createElement('th', ctx.data, ctx.children);
  },
};
</script>
