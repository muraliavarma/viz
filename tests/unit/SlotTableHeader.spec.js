import { mount } from '@vue/test-utils';
import SlotTableHeader from '@/components/SlotTableHeader.vue';

const click = jest.fn();
const wrapper = mount(SlotTableHeader, {
  context: {
    children: ['foo'],
    on: { click },
    class: {
      bar: true,
    },
    attrs: {
      baz: 'quux',
    },
  },
});

describe('SlotTableHeader.vue', () => {
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('triggers click event', () => {
    wrapper.trigger('click');
    expect(click).toHaveBeenCalled();
  });
});
