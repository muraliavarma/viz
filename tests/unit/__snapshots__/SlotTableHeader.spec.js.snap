// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotTableHeader.vue has the expected html structure 1`] = `
<th
  baz="quux"
  class="bar"
>
  foo
</th>
`;
